avalanche/core.py

```diff
@@ -68,6 +78,18 @@ def after_eval(self, strategy: Template, *args, **kwargs) -> CallbackResult:
         """Called after `eval` by the `BaseTemplate`."""
         pass
 
+    def _check_distributed_support(
+            self,
```

Comment:
That is a bit tricky. The problem with an inherited class-level flag is that child classes of a plugin may not actually support distributed training (differently from the father class).

The idea is that the father class sets the default value of the constructor parameter `supports_distributed` to None. Then, 2 situations may happen:

1. If `self` is an instance of a child class
  - If the child constructor passes True/False as a value to `supports_distributed` to the super constructor, ten that value will be used as-is.
  - If it does not pass any value, it means that `_check_distributed_support` is called with `None` and it will detect that the child class did not explicitly support distributed training, returning False.
2. `self` is an instance of the "current" class: `_check_distributed_support` is called with `None`, detects that the father class supports distributed training and returns True.

In this way, for a plugin class to not flag itself as supporting distributed training, it is sufficient to call the super constructor by not passing the `supports_distributed` parameter at all. This is good because we can mark existing plugins as not explicitly supporting distributed training without the need to change their code.

Reply:
Updated in last commit. It now uses this: https://peps.python.org/pep-0487/